### DISCUSSION TITLE
PurgeCss safelist configuration

### DISCUSSION THREAD

**Discussion by MyNameTony1000** [KingSora/OverlayScrollbars#501] [Q&A] [answered]
Hello, I initialize the plugin like this:

const osInstance = OverlayScrollbars(document.getElementsByTagName('body')[0], {
  paddingAbsolute: true,
  overflow: {
    x: 'hidden',
    y: 'scroll',
  },
  scrollbars: {
    visibility: 'auto',
    autoHide: 'scroll',
    autoHideDelay: 1000,
  },
});

with addition to html (data-overlayscrollbars-initialize)
During normal work with the project, everything is fine, although the plugin does not add any extra html to the body, except for the scrollbars themselves and attributes on the html tag. In initialization on internal blocks - adds its markup.
But when I make a build - the plugin does not work on the body, the attribute with html disappears and that's it, but the internal blocks are initialized in the same way as during normal work. Could you help me solve the problem? I can't find the cause of this problem myself.

**Comment by KingSora** [comment, marked as answer]
@MyNameTony1000 Good day :)

That sounds weird, can you create a minimal ecample of your code on jsfiddle, codesandbox or stackblitz?

**Reply by MyNameTony1000** [reply]
It seems to me that even if I can make an example of a project after "build", it will not be clear because there is minification and other points for optimization. In the working version, everything works as it should.

The only thing I noticed is that the data-overlayscrollbars-initialize added initially in the working version does not disappear:

`<html data-overlayscrollbars-initialize data-overlayscrollbars="viewport scrollbarHidden" data-overlayscrollbars-overflow-x="" data-overlayscrollbars-overflow-y="scroll"></html>`

And in the version after "build" this attribute simply disappears and the rest are not added. All other bands work fine. And yet, as far as I understand, despite the fact that the "body" is specified as the initialization element, all attributes are added to "html" and it turns out that it becomes the initialization element. In general, I don’t know) Are there any peculiarities when initializing to the "body" element?

could it be related to this setting:
`cancel: {
    nativeScrollbarsOverlaid: true,
    body: null,
 }`
If it is the default?

**Reply by MyNameTony1000** [reply]
UPDATED PROBLEM: 

This issue occurs due to the use of the purgeCss plugin. Probably not all classes and attributes necessary for initialization are explicitly registered. Have you thought about creating an exception only for classes that are needed during initialization, so that they cannot be reached by such things?

**Reply by KingSora** [reply]
So `purgeCss` is removing required classes from the final css file.. I can't really optimize that because at some point the majority of those classes is needed. I read the documentation of purgeCSS and you could safelist that yourself:
```js
const purgecss = new Purgecss({
    safelist: {
      deep: [/^\.os-.*/, /.*data-overlayscrollbars.*/],
      variables: [/^--os-.*/],
      keyframes: [/^os-.*/],
    }
})
```

**Reply by MyNameTony1000** [reply]
Thank you very much for your help, I'm not strong in (regexp), but this helped:

`
safelist: {
     deep: [ /^os-/, /overlayscrollbars/],
     variables: [/^--os-/],
     keyframes: [/^os-/],
   },
`

With "*" he refused to work normally)

**Reply by KingSora** [reply]
Sweet, glad I could help! :)
